convoluted-converter/_solution/convr.js

```diff
@@ -1,29 +1,28 @@
 const numberSystems = [
-  {name: "bin", prefix: "0b", base: 2, shape: /^0b[01]+$/},
-  {name: "hex", prefix: "0x", base: 16, shape: /^0x[0-9a-fA-F]+$/},
-  {name: "dec", prefix: "", base: 10, shape: /^(0|[1-9]\d*)$/},
+  {name: "bin", prefix: "0b", base: 2, shape: /^0b[01]+$/},
+  {name: "hex", prefix: "0x", base: 16, shape: /^0x[0-9a-f]+$/i},
+  {name: "dec", prefix: "", base: 10, shape: /^(0|[1-9]\d*)$/},
 ];
 
-const normalise = (ns, input) => parseInt(input.substr(ns.prefix.length), ns.base);
-const translate = (ns, intermediate) => ns.prefix + intermediate.toString(ns.base);
-const convert = (targetName, input) => {
-  const inputNS = numberSystems.find(n => n.shape.test(input))
-    || (() => {throw "Input number invalid"})();
-  const targetNS = numberSystems.find(n => n.name === targetName)
-    || (() => {throw "Target option invalid"})();
-  const intermediate = normalise(inputNS, input);
-  return translate(targetNS, intermediate);
+const findNumberSystemByName = name => numberSystems.find(ns => ns.name === name);
+const findNumberSystemByExampleNumber = numberStr => numberSystems.find(ns => ns.shape.test(numberStr));
+const parseNumber = (ns, numberStr) => parseInt(numberStr.substr(ns.prefix.length), ns.base);
+const formatNumber = (ns, number) => ns.prefix + number.toString(ns.base);
+const convert = (targetNumberSystemName, inputNumberStr) => {
+  const inputNS = findNumberSystemByExampleNumber(inputNumberStr) || fail("Input number invalid");
+  const targetNS = findNumberSystemByName(targetNumberSystemName) || fail("Target option invalid");
+  const inputNumber = parseNumber(inputNS, inputNumberStr);
+  return formatNumber(targetNS, inputNumber);
 };
+const fail = message => { throw message };
```

Comment:
Probably the better way to go indeed – there had been more critical feedback on the IIFE style…